utils/linkage.py

```diff
@@ -106,4 +25,22 @@ def get_street_from_address_line_1(address_line_1: str) -> str:
         ...
     ValueError: address_line_1 is PO Box
     """
-    pass
+
+    if not address_line_1 or address_line_1.isspace():
+        raise ValueError("address_line_1 must have whitespace")
+
+    address_line_lower = address_line_1.lower()
+
+    if "po box" in address_line_lower:
+        raise ValueError("address_line_1 is PO Box")
+
+    parts = address_line_1.split()
+
+    string = []
+    for part in parts:
+        if part.isdigit() or "." in part:
```

Comment:
handle case discussed in meeting of "Uber St.", "3NW 59th St", "300 59 St" -- using nayna's package may be helpful